osquery/tables/system/windows/shortcut_files.cpp

```diff
@@ -0,0 +1,219 @@
+ * root directory of this source tree.
+ *
+ * SPDX-License-Identifier: (Apache-2.0 OR GPL-2.0-only)
+ */
+
+#include <osquery/core/core.h>
+#include <osquery/core/tables.h>
+#include <osquery/filesystem/filesystem.h>
+#include <osquery/tables/system/windows/registry.h>
+#include <osquery/utils/conversions/join.h>
+
+#include <osquery/logger/logger.h>
+#include <osquery/utils/windows/shelllnk.h>
+
+#include <boost/filesystem.hpp>
+
+#include <sstream>
+#include <string>
+
+namespace osquery {
+namespace tables {
+
+QueryData genShortcutFiles(QueryContext& context) {
+  QueryData results;
+  auto paths = context.constraints["path"].getAll(EQUALS);
+  // Expand contraints
+  context.expandConstraints(
+      "path",
+      LIKE,
+      paths,
+      ([&](const std::string& pattern, std::set<std::string>& out) {
+        std::vector<std::string> patterns;
+        auto status =
+            resolveFilePattern(pattern, patterns, GLOB_ALL | GLOB_NO_CANON);
+        if (status.ok()) {
+          for (const auto& resolved : patterns) {
+            out.insert(resolved);
+          }
+        }
+        return status;
+      }));
+
+  boost::system::error_code ec;
+  for (const auto& lnk : paths) {
+    boost::filesystem::path path = lnk;
+    if (!boost::filesystem::is_regular_file(path, ec)) {
+      continue;
+    }
+    std::string lnk_content;
+    if (!readFile(path, lnk_content, 20).ok()) {
+      LOG(WARNING) << "Failed to read shortcut file: " << lnk;
+      continue;
+    }
+    std::stringstream check_ss;
+    for (const auto& hex_char : lnk_content) {
+      std::stringstream value;
+      value << std::hex << std::uppercase << (int)(unsigned char)(hex_char);
+      // Add additional 0 if single hex value is 0-F
+      if (value.str().size() == 1) {
+        check_ss << "0";
+      }
+      check_ss << value.str();
+    }
+    std::string header_sig = check_ss.str();
+
+    // Check for shortcut header size (0x0000004c) and GUID
+    // (00021401-0000-0000-c000-000000000046)
+    if (header_sig != "4C0000000114020000000000C000000000000046") {
+      continue;
+    }
+
+    // Read the whole shortcut file
+    lnk_content = "";
+    if (!readFile(path, lnk_content).ok()) {
+      LOG(WARNING) << "Failed to read shortcut file: " << lnk;
+      continue;
+    }
+    std::stringstream ss;
+    for (const auto& hex_char : lnk_content) {
+      std::stringstream value;
+      value << std::hex << std::uppercase << (int)(unsigned char)(hex_char);
+      // Add additional 0 if single hex value is 0-F
+      if (value.str().size() == 1) {
+        ss << "0";
+      }
+      ss << value.str();
+    }
+
+    const std::string lnk_hex = ss.str();
+    LinkFileHeader data;
+    TargetInfo target_data;
+    LocationInfo location_data;
+    ExtraDataTracker extra_data;
+    DataStringInfo data_info_string;
+
+    data = parseShortcutHeader(lnk_hex);
+    if (data.header == "") {
+      continue;
+    }
+    std::string data_string = lnk_hex.substr(152);
```

Comment:
Prefer this 152 to be a constant with a meaningful label.

Reply:
good idea
added variable